### PR TITLE
Format numbers properly.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: jit-compiler/src/codegen.rs [hunk 1]
```diff
@@ -290,9 +300,18 @@ impl<'a, T: FieldElement> CodeGenerator<'a, T> {
     }
 }
 
-pub fn escape_symbol(s: &str) -> String {
-    // TODO better escaping
-    s.replace('.', "_").replace("::", "_")
+fn format_number(n: &BigUint) -> String {
```
Comment (Schaeff) [inline jit-compiler/src/codegen.rs:303]:
```suggestion
fn format_integer(n: &BigUint) -> String {
```
?

File: jit-compiler/src/codegen.rs [hunk 2]
```diff
@@ -211,6 +218,9 @@ impl<'a, T: FieldElement> CodeGenerator<'a, T> {
                     BinaryOperator::ShiftLeft => {
                         format!("(({left}).clone() << u32::try_from(({right}).clone()).unwrap())")
                     }
+                    BinaryOperator::ShiftRight => {
+                        format!("(({left}).clone() >> usize::try_from(({right}).clone()).unwrap())")
```
Comment (Schaeff) [inline jit-compiler/src/codegen.rs:222]: `ShiftLeft` above uses `u32::try_from`, would be good to have consistency

File: jit-compiler/src/codegen.rs [hunk 3]
```diff
@@ -175,12 +179,15 @@ impl<'a, T: FieldElement> CodeGenerator<'a, T> {
...
-                    _ => unreachable!(),
+                if *type_ == Type::Int {
+                    format_number(value)
+                } else {
+                    let value = u64::try_from(value).unwrap_or_else(|_| unimplemented!());
+                    match type_ {
+                        Type::Fe => format!("FieldElement::from({value}_u64)"),
+                        Type::Expr => format!("Expr::from({value}_u64)"),
+                        _ => unreachable!(),
+                    }
```
Comment (Schaeff) [inline jit-compiler/src/codegen.rs:190]: I find having a single match for all types clearer

File: jit-compiler/src/codegen.rs (hunk 2 above)
Comment (chriseth) [reply jit-compiler/src/codegen.rs:222]: true, right somehow didn't work with u32, will check.

File: jit-compiler/src/codegen.rs (hunk 3 above)
Comment (chriseth) [reply jit-compiler/src/codegen.rs:190]: so you would prefer doubling line 185?

Comment (Schaeff) [reply jit-compiler/src/codegen.rs]: Yes

Comment (chriseth) [reply jit-compiler/src/codegen.rs]: Done.

**Review by lvella — COMMENTED** [review]
LGTM

File: jit-compiler/src/codegen.rs (hunk 2 above)
Comment (chriseth) [reply jit-compiler/src/codegen.rs:216]: Changed both to usize, since this is what ibig implements.

**Review by Schaeff — APPROVED** [review]

**Pull request merged**